core/templates/pages/exploration-editor-page/services/exploration-rights.service.ts

```diff
@@ -230,6 +230,14 @@ export class ExplorationRightsService {
   }
 
   removeRoleAsync(memberUsername: string): Promise<void> {
+    const categories = ['ownerNames', 'editorNames', 'viewerNames'] as const;
+    let initialState: {[key: string]: string[]} = {};
+    categories.forEach(category => {
+      if (this[category]?.includes(memberUsername)) {
```

Comment:
Why the question mark? If we want to be careful about handling the case where the category is non-existent then an error should be thrown if we don't expect that case to happen.

Reply:
The ? is basically due to some of the unit tests not defining a category, but this doesn't mean an error has to be thrown. For instance, if only ers.editorNames is initialized, and then removeRoleAsync is called, ers.ownerNames and ers.viewerNames were never initalized. So when this if check runs, if its not defined, it will just skip over that category and check the next one for the username. In practice, on the UI side, I think that all the user categories are initialized. This statement was initially if(this[category] && this[category].includes[memberUsername]), which is equivalent.

Reply:
Hm, I think in that case it would be better if you had the unit tests define whatever categories are necessary, and removed the "?".

An alternative is to have `categories` be defined more globally in this serivce, e.g. change this.ownerNames to this.roles.ownerNames and similarly for the others, and define `categories` as the keys of `this.roles`. Then tests can specify whatever they like for this.roles and it should "just work". But this might be more of a refactor than you'd want to take on, in which case please see the first paragraph.

Reply:
So it appears that some of the other frontend tests failed when I removed the "?". Specifically, in settings-tab.component.spec.ts, as some of the tests call removeRole, for instance ,

component.removeRole('username', 'editor'); 

removeRole will then call removeRoleasync, and since only the editor category was created here originally, this causes the same error to be thrown (ownerNames, viewerNames would be undefined). Would it be simpler to just keep the "?" instead, as I'm not sure I could define these categories before component.removeRole() is called.

Reply:
No, I don't think we should be making that change just for tests. Why can't you define the necessary categories in the test?

Reply:
Fixed